src/components/card/Card.jsx

```diff
@@ -0,0 +1,81 @@
+import "./card.css";
+import { useNavigate } from "react-router-dom";
+import { useAuth, useWishList } from "../../context";
+export function Card({ product }) {
+  const { user } = useAuth();
+  const navigate = useNavigate();
+  const { wishList, addToWishList, removeFromWishList } = useWishList();
+
+  const productInWishList = wishList?.some((p) => p._id === product._id)
+    ? true
+    : false;
+  return (
+    <div className="sm-card">
+      <div className="sm-card-img">
+        {product.badgeName && (
+          <span className="sm-card-badge">{product.badgeName}</span>
+        )}
+        <button
+          className="sm-card-fav"
+          title={`${
+            productInWishList && user
+              ? "Remove From Wishlist"
+              : "Add To Wishlist"
+          }`}
+          onClick={() => {
+            if (!user) {
+              navigate("/auth");
+            } else if (productInWishList) {
+              removeFromWishList(product);
+            } else {
+              addToWishList(product);
+            }
+          }}
+        >
+          <i
+            className={`fa-heart ${productInWishList && user ? "fas" : "far"}`}
+          ></i>
+        </button>
+        <img src={product.image} alt={product.imageAlt} />
+      </div>
+      <div className="sm-card-content">
+        <h2 className="sm-card-content-title" title="Click To Know More!!">
+          {product.title}
+        </h2>
+        <h2 className="sm-card-content-brand" title="Click To Know More!!">
+          <p className="sm-card-flex-rating">
+            {`by `}&nbsp;
+            <a href="#" className="link-brand">
```

Comment:
Why anchor tag avoid them react might give you a warning

Reply:
Will update it